File: run/bak/cpp/ffi/Test/LazyEvaluationFFICheatcode.cc
#include "purescript.h"
namespace { volatile int dummy; }
FOREIGN_BEGIN( Test_LazyEvaluationFFICheatcode )
exports["runLazyEvaluationFFICheatcode"] = [](const boxed& in) -> boxed { 
    int n = unbox<int>(in); int acc = 0;
    for(int i=0; i<n; i++) acc += 1;
    dummy = acc; return acc;
};
FOREIGN_END

File: scratch/gopurs-adt-reuse-validation-20260916/before-pure/src/Test/LazyEvaluationFFICheatcode.php
<?php
$exports['runLazyEvaluationFFICheatcode'] = function($limit) {
    $n = (int)$limit * 1000; // 1000 * 1000 = 1000000
    $acc = 0;
    for ($i = 0; $i < $n; $i++) {
        $acc += 1;
    }
    return $acc;
};
return $exports;
